lib/include/tree_sitter/api.h

```diff
@@ -134,10 +134,30 @@ typedef enum {
   TSQueryErrorLanguage,
 } TSQueryError;
 
+typedef struct {
+  void *(*malloc)(size_t);
+  void *(*calloc)(size_t, size_t);
+  void *(*realloc)(void *, size_t);
+  void (*free)(void *);
+} TSAllocator;
```

Comment:
I'd prefer to remove this struct definition from the public API, and just pass the function pointers as 4 separate parameters to `ts_set_allocator`.

Then, I would just store each function in its own `static` variable, declared in `alloc.c`. That will remove a small amount of indirection when invoking the functions.

Reply:
If the variable is defined as `static`, it cannot be invoked in other source files. Perhaps it should be a normal global variable?

Reply:
Oh, I was thinking that it'd be internal to `alloc.c`, and exposed via some functions in `alloc.h`, but that's not necessary. What you did with `ts_current_alloc` looks great. Thanks.